aws/aws-parallelcluster-cookbook
# [headnode metrics] add headnode disk and memory metrics to CloudWatch agent configuration

### Description of changes
* Add new CloudWatch metrics in order to monitor memory usage and disk usage of the root volume on head node.
* Use "InstanceId" as append dimension, so that we can use these metrics later by instance id for downstream usage of the metrics such as visualization and alarms.
* Use "InstanceId, path" as an aggregation dimension so that we can use InstanceId and path to specify a disk metric without the need to retrieve device and file system type information.

### Tests
Manual tests: 

After code change, create a new cluster, and validate that new metrics (under the "CWAgent" namespace) show up in CloudWatch as expected:

 * Under the "InstanceId, device, fstype, path" dimension
    * disk_used_percent
 * Under the "InstanceId" dimension
    * mem_used_percent
  * Under the "InstanceId, path" dimension (aggregated)
    * disk_used_percent 

For a head node of c5.large Nitro instance, confirmed that the "reboot" of the head node didn't affect the metrics.


### References
* [PR to add CloudWatch Dashboard graphs based on these metrics](https://github.com/aws/aws-parallelcluster/pull/4869)
* [PR to add CloudWatch alarms based on these metrics](https://github.com/aws/aws-parallelcluster/pull/4877)

### Checklist

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
